src/Compilers/Test/Core/ModuleInitializer.cs

```diff
@@ -17,6 +17,10 @@ internal static void Initialize()
         {
             Trace.Listeners.Clear();
             Trace.Listeners.Add(new ThrowingTraceListener());
+
+            // Make sure we load DSRN from the directory containing the unit tests and not from a runtime directory on .NET 5+.
+            Environment.SetEnvironmentVariable("MICROSOFT_DIASYMREADER_NATIVE_ALT_LOAD_PATH", Path.GetDirectoryName(typeof(ModuleInitializer).Assembly.Location));
+            Environment.SetEnvironmentVariable("MICROSOFT_DIASYMREADER_NATIVE_USE_ALT_LOAD_PATH_ONLY", "1");
```

Comment:
I'm a bit worried about modifying the environment from a test. Smells like a concurrency issue waiting to happen. Is that something we do elsewhere? 
Tagging @jaredpar to advise.

Reply:
Looking around for other uses of `Environment.SetEnvironmentVariable` for tests, I see that do have precedent in `TestRunner.RunAllOnHelixAsync`. Should we move setting the two new environment variables there?

Reply:
That would break running tests in VS.

Reply:
Ah, right.
Just to clarify, does xunit or any other mechanism prevent this environment variable change from affecting the machine at large?

Reply:
Not sure what you mean. The environment variable is set per-process, not per-machine.

Reply:
`Environment.SetEnvironmentVariable` can modify the machine environment, but it looks like that uses a different overload (with `EnvironmentVariableTarget` parameter).
I'm good then. Thanks

Reply:
> I'm a bit worried about modifying the environment from a test. Smells like a concurrency issue waiting to happen. Is that something we do elsewhere?

In general this is true. For example if we even see environment variables being manipulated from within individual unit tests then that is highly suspicious. In this case though we are manipulating an the module initializer level so that effectively means it's setting it at the process level. 

The one part I'm worried about though is there is the potential for a subtle timing issue. Essentially imagine if DiaSym gets loaded before the module initializer runs. Then it will potentially load from the wrong place and there is no protection or detection for that. 

@tmat in the case this does happen how would we know DiaSym got loaded from the wrong place?

Reply:
If the wrong one gets loaded the CompilerInfo test fails. Everything else should be ok. 
BTW, this is temporary - once we insert the RTM one into .NET runtime and we start running on that runtime we can remove the env variable.

Reply:
Do we have an issue tracking that removal I can follow?

Reply:
Filed https://github.com/dotnet/roslyn/issues/57526